PHP 8.0 Upgrade

aidan-casey opened the pull request:
Not entirely sure what the appetite for change is here, but thought I would start a pull request to see if it was welcomed.

This PR does the following:

- Enforces parameter types through modern PHP type hinting thus removing the need for some assertion classes.
- Enforces property types through modern PHP type hinting.
- Removes unnecessary doc blocks.
- Updates directory structure to match Laravel recommendations.
- Adds tests for 100% code coverage.
- Requires PHP 8.0.
- Updates test workflow to run in about 13 seconds as opposed to 6 minutes.

denisdulici commented:
I follow [Laravel](https://laravel.com/docs/9.x/releases#support-policy) for the PHP version requirement, so can't merge this PR as is.

aidan-casey commented:
@d

aidan-casey commented:
@denisdulici if I make this PHP 8.0 compatible would you be of the mind to merge?

denisdulici commented:
Sure, why not.

aidan-casey commented:
@denisdulici - I have reverted the PHP 8.1 specific changes so that this will support PHP 8 and 8.1. Note that I have also updated the directory structure to match Laravel's recommendations and update the testing workflow to gain some faster run times.

denisdulici commented:
Thanks for your effort. Can you please keep the `Laravel 8` compatibility? At least until Laravel [drops](https://laravel.com/docs/releases#support-policy) the bug fixes.

denisdulici commented:
No, they're not the same thing, and this package requires at least `PHP 8` + `Laravel 8` since the 2.0 version.

aidan-casey commented:
@denisdulici - Sorry I misunderstood based on your comment earlier:

> I follow [Laravel](https://laravel.com/docs/9.x/releases#support-policy) for the PHP version requirement, so can't merge this PR as is.

I have updated the dependencies as requested and also added them to the testing workflow to ensure everything is in order.

denisdulici commented on tests/TestCase.php:
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Akaunting\Tests\Money;
```
The namespace must be `Akaunting\Money\Tests`. Same issue for all test classes.

aidan-casey replied:
Fixed.

aidan-casey commented on src/Provider.php:
```diff
@@ -8,15 +8,10 @@
     {
         $this->publishes([
-            __DIR__ . '/Config/money.php' => config_path('money.php'),
+            __DIR__ . '/../money.php' => config_path('money.php'),
```
My bad, good catch! This is resolved in the latest commit.

denisdulici commented:
Thanks @aidan-casey for this huge contribution 🙏

aidan-casey commented:
No problem. Thanks for the feedback throughout!
